packages/indexer-client/src/providers/envioIndexerClient.ts

```diff
@@ -75,67 +75,72 @@ export class EnvioIndexerClient implements IIndexerClient {
     }
 
     /** @inheritdoc */
-    async getEventsBySrcAddress(params: {
-        chainId: ChainId;
-        srcAddresses: Address[];
-        from?: {
-            blockNumber?: number;
-            logIndex?: number;
-        };
-        to: {
-            blockNumber: number;
-            logIndex: number;
-        };
-        limit?: number;
-    }): Promise<AnyIndexerFetchedEvent[]> {
+    async getEvents(params: GetEventsFilters): Promise<AnyIndexerFetchedEvent[]> {
         try {
             const { chainId, srcAddresses, from, to, limit = 100 } = params;
-            const { blockNumber: toBlock, logIndex: toLogIndex } = to;
-            const { blockNumber: fromBlock, logIndex: fromLogIndex } = from ?? {
-                blockNumber: 0,
-                logIndex: 0,
-            };
+
+            // Build the _and conditions array
+            const andConditions = [];
+            andConditions.push(`chain_id: { _eq: $chainId }`);
+            const vars: Record<string, unknown> = { chainId };
+
+            // Add srcAddresses filter if provided
+            if (srcAddresses && srcAddresses.length > 0) {
+                andConditions.push(`src_address: { _in: $srcAddresses }`);
+                vars["srcAddresses"] = srcAddresses;
+            }
+
+            if (from != undefined && from != null) {
+                andConditions.push(`
+                    _or: [
+                        { block_number: { _gt: $fromBlock } },
+                        {
+                            _and: [
+                                { block_number: { _eq: $fromBlock } },
+                                { log_index: { _gt: $fromLogIndex } }
+                            ]
+                        }
+                    ]
```

Comment:
This might not be relevant as you might be using this provider in a super specific way but how would you get all events from a single block?

I can only think of something like this:
```
from: { blockNumber: x, logIndex: 0 }
to: { blockNumber: x + 1, logIndex: 0 }
```

But it misses the log 0 from block x and includes log 0 from block x + 1.

Reply:
didn't think of this case since we fetch events individually (blockNumber and logIndex), not by blocks but i get your point. i will add it to tech debt